src/Rector/Deprecation/ControllerBaseLRector.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace DrupalRector\Rector\Deprecation;
+
+use DrupalRector\Utility\TraitsByClassHelperTrait;
+use PhpParser\Node;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * Replaces deprecated \Drupal\Core\Routing\LinkGeneratorTrait::l() calls.
+ *
+ * See https://www.drupal.org/node/2614344 for change record.
+ *
+ * What is covered:
+ * - Trait usage when the `LinkGeneratorTrait` is already present on the class
+ *
+ * Improvement opportunities
+ * - Add trait for classes
+ *   - `use LinkGeneratorTrait;`
+ */
+final class ControllerBaseLRector extends AbstractRector
```

Comment:
@Kbentham, can we name this (and the examples) `LinkGeneratorTraitL`, etc? I think this started as only dealing with the ControllerBase class, but now includes more.